src/RequestHandlerSwooleRunner.php

```diff
@@ -193,24 +206,101 @@ function (Throwable $exception) use ($serverRequestErrorResponseGenerator) : Res
             ));
         }
 
-        $this->logger = $logger ?: new StdoutLogger();
+        $this->logger = $logger ? : new StdoutLogger();
+        $this->pidManager = $pidManager;
     }
 
     /**
      * Run the application
      */
     public function run() : void
     {
-        $this->swooleHttpServer->on('start', [$this, 'onStart']);
-        $this->swooleHttpServer->on('request', [$this, 'onRequest']);
-        $this->swooleHttpServer->start();
+        $opts = new Getopt([
+            'daemonize|d'  => 'daemonize the swoole server process',
+            'worker_num|n=i' => 'set worker nums',
+            'dispatch_mode=i' => 'set dispatch_mode',
+        ]);
+        $args = $opts->getArguments();
+        $action = $args[0] ?? null;
+        switch ($action) {
+            case 'stop':
+                $this->stopServer();
+                break;
+            case 'start':
+            default:
+                $this->startServer([
+                    'daemonize' => $opts->getOption('daemonize') ? (bool) $opts->getOption('daemonize') : false,
+                    'dispatch_mode' => $opts->getOption('dispatch_mode') ? (int) $opts->getOption('dispatch_mode') : 3,
+                    'worker_num' => $opts->getOption('worker_num') ? (int) $opts->getOption('worker_num') : 1,
+                ]);
+                break;
+        }
+    }
+
+    /**
+     * Start swoole server
+     *
+     * @param array $options Swoole server options
+     */
+    public function startServer(array $options = []) : void
+    {
+        $swooleServer = $this->server->createSwooleServer($options);
+        $swooleServer->on('start', [$this, 'onStart']);
+        $swooleServer->on('request', [$this, 'onRequest']);
+        $swooleServer->start();
+    }
+
+    /**
+     * Stop swoole server
+     */
+    public function stopServer() : void
+    {
+        if (! $this->isRunning()) {
+            $this->logger->info('Server is not running yet');
+            return;
+        }
+        [$masterPid, ] = $this->pidManager->read();
+        $this->logger->info('Server stopping ...');
+        $result = SwooleProcess::kill((int)$masterPid);
```

Comment:
Here and elsewhere: space after casting operator and before variable name, please (i.e., `(int) $masterPid`, instead of `(int)$masterPid`).